### ISSUE TITLE
Can't access current (previous) project in before-switch-project-hook

### ISSUE THREAD

**Issue by pwalsh** [bbatsov/projectile#1042]
## Expected behavior

When invoking a function via a `projectile-before-switch-project-hook`, I'd expect `projectile-project-name` to return the project name from *before* the switch, or `nil` if no project is active.

## Actual behavior

When invoking a function via a `projectile-before-switch-project-hook`, `projectile-project-name` returns the project name for the project being switched to.

I assume, that this is not a bug, but rather that, by design, `projectile-before-switch-project-hook` does not mean what I expect it to mean based on the name. In that case, is there another hook that would achieve such behaviour?

## Steps to reproduce the problem

```
  (defun before-switch-project ()
    "Actions on closing a project."
    (message (concat (projectile-project-name) ": before switch"))
    ;; kill the buffer based on names generated by projectile-run-eshell
    (if (get-buffer (concat "*eshell " (projectile-project-name) "*"))
        (kill-buffer (concat "*eshell " (projectile-project-name) "*"))
      )
    )

  (add-hook 'projectile-before-switch-project-hook #'before-switch-project)
```

## Environment & Version information

- Mac OSX
- Projectile version: 20160728.52
- Spacemacs version: 0.105.21
- Emacs version: 24.5.1

**Comment by bbatsov** [comment]
Fixed by #1051.

**Comment by pwalsh** [comment]
@bbatsov nope. now `projectile-project-name` returns `-` in all cases.
